tests/library/status.py

```diff
@@ -0,0 +1,113 @@
+from adcm_pytest_plugin.docker_utils import ADCM
+
+from tests.library.utils import RequestFailedException, get_json_or_text
+
+POSITIVE_STATUS = 0
+_PATH_TO_ADCM_SECRETS = '/adcm/data/var/secrets.json'
+
+
+class ADCMObjectStatusChanger:
+    """
+    Allows you to "change" status of dummy Components and Hosts.
+
+    If you want to "enable" Cluster or Service,
+        you should set positive status on all hosts and components on it.
+    If you want to "disable" Cluster or Service,
+        you should set negative status at least one of its components or hosts.
+
+    After using status-changing methods you'll need to call `reread` method
+        to be able to get newly set status.
+
+    Consider that status != 16 will be reset to 16 after timeout (300 seconds).
+    """
+
+    def __init__(self, client: ADCMClient, container: ADCM):
+        self.client = client
+        api_root = client.url
+        self.host_url_template = f"{api_root}/status/api/v1/host/" + "{}/"
+        self.component_url_template = self.host_url_template + 'component/{}/'
+        self._headers = {
+            'Authorization': f'Token {self._extract_status_api_token(container)}',
+            'Content-Type': 'application/json',
+        }
+
+    def enable_cluster(self, cluster: Cluster) -> None:
+        """Enable cluster by setting positive status for each host and component"""
+        hosts = set()
+        for hostcomponent in cluster.hostcomponent():
+            host_id, component_id = hostcomponent['host_id'], hostcomponent['component_id']
+            self._set_component_status(host_id, component_id, POSITIVE_STATUS)
+            if host_id not in hosts:
+                self._set_host_status(host_id, POSITIVE_STATUS)
+                hosts.add(host_id)
+
+    def set_host_positive_status(self, hosts: Collection[Host]) -> None:
+        """Make hosts "working" by setting status to 0"""
+        for host in hosts:
+            self._set_host_status(host.id, POSITIVE_STATUS)
+
+    def set_host_negative_status(self, hosts: Collection[Host], status: int = 16) -> None:
+        """Make hosts "not working" by setting status != 0"""
+        for host in hosts:
+            self._set_host_status(host.id, status)
+
+    def set_component_positive_status(self, host_components: Collection[Tuple[Host, Component]]) -> None:
+        """Make components "working" by setting status to 0"""
+        for host, component in host_components:
+            self._set_status(self.host_url_template.format(host.id, component.id), POSITIVE_STATUS)
+
+    def set_component_negative_status(
+        self, host_components: Collection[Tuple[Host, Component]], status: int = 16
+    ) -> None:
+        """Make components "not working" by settings status != 0"""
+        for host, component in host_components:
+            self._set_status(self.host_url_template.format(host.id, component.id), status)
+
+    def _set_host_status(self, host_id: int, status: int) -> None:
```

Comment:
this function will be useful in public scope with `Host` argument type, not id
same for component

Reply:
@skhomuti 
Do you mean it'll be useful to be able to pass a single host instead of a collection?
This particular function is made to be used in both cases: when we do not have a host/component objects and when we do.
I can make set status functions to take either collection or a single value, what do you think?

Reply:
yep, collection or a single value in arguments will be ok

Reply:
then naming will be "set_host_*" and "set_component_*" I think